### PR TITLE
Remove WebRTC APIs that has never been supported

### PR DESCRIPTION
**Pull request by saschanaz**
#### Summary

Splitted from #6854 and covers only WebRTC changes. @foolip

#### Related issues

#6809

### CODE REVIEW

File: api/RTCCertificate.json [hunk 1]
```diff
@@ -143,54 +143,6 @@
             "deprecated": false
           }
         }
-      },
-      "getSupportedAlgorithms": {
-        "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/RTCCertificate/getSupportedAlgorithms",
```
Comment (foolip) [inline api/RTCCertificate.json:149]: This is 404 👍

File: api/RTCCertificate.json [hunk 2]
```diff
@@ -143,54 +143,6 @@
             "deprecated": false
           }
         }
-      },
-      "getSupportedAlgorithms": {
```
Comment (foolip) [inline api/RTCCertificate.json:147]: Confirmed not in Chromium, Gecko, or WebKit, just a FIXME in WebKit:
https://github.com/WebKit/WebKit/blob/751a4873a15c12356e413dc947dd29d481889a2c/Source/WebCore/Modules/mediastream/RTCCertificate.idl#L45

File: api/RTCIceTransport.json [hunk 3]
```diff
@@ -243,56 +243,6 @@
           }
         }
       },
-      "gatheringstatechange_event": {
```
Comment (foolip) [inline api/RTCIceTransport.json:246]: Looks like the event is in Chromium, Gecko and WebKit source and `ongatheringstatechange` has support data:
https://github.com/mdn/browser-compat-data/blob/820c07582578aab9b125ef8923d3b11f3f108cc1/api/RTCIceTransport.json#L548-L596

Just copying that data would be a good start to show that this shouldn't be removed.

File: api/RTCIceTransport.json [hunk 4]
```diff
@@ -741,56 +691,6 @@
           }
         }
       },
-      "selectedcandidatepairchange_event": {
```
Comment (foolip) [inline api/RTCIceTransport.json:744]: https://github.com/mdn/browser-compat-data/blob/820c07582578aab9b125ef8923d3b11f3f108cc1/api/RTCIceTransport.json#L597-L645

File: api/RTCIceTransport.json [hunk 5]
```diff
@@ -888,56 +788,6 @@
           }
         }
       },
-      "statechange_event": {
```
Comment (foolip) [inline api/RTCIceTransport.json:891]: https://github.com/mdn/browser-compat-data/blob/820c07582578aab9b125ef8923d3b11f3f108cc1/api/RTCIceTransport.json#L646-L694

File: api/RTCRemoteOutboundRtpStreamStats.json [hunk 6]
```diff
@@ -145,55 +145,6 @@
             "deprecated": false
           }
         }
-      },
-      "reportsSent": {
```
Comment (foolip) [inline api/RTCRemoteOutboundRtpStreamStats.json:149]: `RTCRemoteOutboundRtpStreamStats` is a dictionary and tricky to test, but I can't find any trace of this in Chromium, Gecko, or WebKit source outside of tests, so I think it should indeed be removed.

File: api/RTCRemoteOutboundRtpStreamStats.json [hunk 7]
```diff
@@ -145,55 +145,6 @@
...
-        "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/RTCRemoteOutboundRtpStreamStats/reportsSent",
```
Comment (foolip) [inline api/RTCRemoteOutboundRtpStreamStats.json:151]: This page exists and will need deleting, presumably.

File: api/RTCIceTransport.json (hunk 4 above)
Comment (saschanaz) [reply api/RTCIceTransport.json:744]: I wonder there should be a way to do the sync automatically, or at least a linter for this 🤔

**Comment by jonjarry** [comment]
YOU GUYS ARE DIRECTLY COMPLICITE AND ARE FACING LITIGATION


On May 23, 2021 at 4:22 PM, Philip Jägenstedt ***@***.***> wrote:


@foolip commented on this pull request.
In api/RTCCertificate.json:
> @@ -143,54 +143,6 @@

             "deprecated": false

           }

         }

-      },

-      "getSupportedAlgorithms": {

-        "__compat": {

-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/RTCCertificate/getSupportedAlgorithms",

This is 404 👍
In api/RTCCertificate.json:
> @@ -143,54 +143,6 @@

             "deprecated": false

           }

         }

-      },

-      "getSupportedAlgorithms": {

Confirmed not in Chromium, Gecko, or WebKit, just a FIXME in WebKit:
https://github.com/WebKit/WebKit/blob/751a4873a15c12356e413dc947dd29d481889a2c/Source/WebCore/Modules/mediastream/RTCCertificate.idl#L45
In api/RTCIceTransport.json:
> @@ -243,56 +243,6 @@

           }

         }

       },

-      "gatheringstatechange_event": {

Looks like the event is in Chromium, Gecko and WebKit source and ongatheringstatechange has support data:
https://github.com/mdn/browser-compat-data/blob/820c07582578aab9b125ef8923d3b11f3f108cc1/api/RTCIceTransport.json#L548-L596
Just copying that data would be a good start to show that this shouldn't be removed.
In api/RTCIceTransport.json:
> @@ -741,56 +691,6 @@

           }

         }

       },

-      "selectedcandidatepairchange_event": {

https://github.com/mdn/browser-compat-data/blob/820c07582578aab9b125ef8923d3b11f3f108cc1/api/RTCIceTransport.json#L597-L645
In api/RTCIceTransport.json:
> @@ -888,56 +788,6 @@

           }

         }

       },

-      "statechange_event": {

https://github.com/mdn/browser-compat-data/blob/820c07582578aab9b125ef8923d3b11f3f108cc1/api/RTCIceTransport.json#L646-L694
In api/RTCRemoteOutboundRtpStreamStats.json:
> @@ -145,55 +145,6 @@

             "deprecated": false

           }

         }

-      },

-      "reportsSent": {

RTCRemoteOutboundRtpStreamStats is a dictionary and tricky to test, but I can't find any trace of this in Chromium, Gecko, or WebKit source outside of tests, so I think it should indeed be removed.

**Comment by jonjarry** [comment]
YUOU HAVE JUST BEEN REPORTED TO THE POLICE 

On May 23, 2021 at 4:24 PM, Kagami Sascha Rosylight ***@***.***> wrote:


@saschanaz commented on this pull request.
In api/RTCIceTransport.json:
> @@ -741,56 +691,6 @@

           }

         }

       },

-      "selectedcandidatepairchange_event": {

I wonder there should be a way to do the sync automatically, or at least a linter for this 🤔

File: api/RTCIceTransport.json [hunk 8]
```diff
@@ -250,13 +250,13 @@
           "description": "<code>gatheringstatechange</code> event",
           "support": {
             "chrome": {
-              "version_added": false
+              "version_added": "75"
```
Comment (foolip) [inline api/RTCIceTransport.json:253]: I can confirm this, it was implemented in https://source.chromium.org/chromium/chromium/src/+/d78d9f90c87a37772bbc55a72fb085d2c86faa85 before the whole feature was enabled, and thus shipped with the rest of `RTCIceTransport`.

File: api/RTCRemoteOutboundRtpStreamStats.json (hunk 7 above)
Comment (foolip) [reply api/RTCRemoteOutboundRtpStreamStats.json:151]: I've sent https://github.com/mdn/content/pull/5787.

**Review by foolip — APPROVED** [review]

**Pull request merged**